core/enumerable.rbs

```diff
@@ -1846,7 +1846,7 @@ module Enumerable[unchecked out Elem] : _Each[Elem]
   #     e = (1..3).chain([4, 5])
   #     e.to_a #=> [1, 2, 3, 4, 5]
   #
-  def chain: (*self enumerables) -> ::Enumerator::Chain[Elem]
+  def chain: (*Enumerable[Elem] enumerables) -> ::Enumerator::Chain[Elem]
```

Comment:
Looks like it takes anything that defines `#each`.
```suggestion
  def chain: (*_Each[Elem] enumerables) -> ::Enumerator::Chain[Elem]
```

maybe even?
```suggestion
  def chain: [Elem2] (*_Each[Elem2] enumerables) -> ::Enumerator::Chain[Elem | Elem2]
```

Reply:
Good point! I adopted the 2nd suggestion :-)